### PR TITLE
fix walkthrough erroring on Pending pods

### PR DESCRIPTION
**Pull request by kibbles-n-bytes**
Fixes flake on #648 

Closes: #724

### CODE REVIEW

File: contrib/hack/test_walkthrough.sh [hunk 1]
```diff
@@ -135,6 +135,10 @@ retry -n 10 \
 
 echo 'Waiting on pods to come up...'
 
+wait_for_expected_output -x -e 'Pending' -n 10 \
```
Comment (pmorie) [inline contrib/hack/test_walkthrough.sh]: I feel like it is better to shoot for 'ready'; there's a chance you could miss something in a pending state and this wouldn't trigger.

Comment (kibbles-n-bytes) [inline contrib/hack/test_walkthrough.sh]: @pmorie It's already doing that; this and the `ContainerCreating` status checks are just to wait until we're in a non-creating/pending state, and then upon reaching that on line 146 we check if the state we ended up in is actually `Ready`.

Comment (pmorie) [inline contrib/hack/test_walkthrough.sh]: @kibbles-n-bytes I don't think I understand what the line this comment is attached to is doing then - can you clarify?

Comment (kibbles-n-bytes) [inline contrib/hack/test_walkthrough.sh]: @pmorie It's waiting for none of the the pods to have a `Pending` state. The `-x` flag for this function makes it wait for the absence of the provided string. Similarly, 142 is waiting for the absence of `ContainerCreating`.

After none of the pods are in either of these states, we then check to see if each of them are actually `Ready` on line 146.

**Pull request merged**